src/PixiEditor/NotifyableObject.cs

```diff
@@ -56,4 +65,24 @@ protected bool SetProperty<T>(ref T backingStore, T value, out T oldValue, Actio
         PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
         return true;
     }
+
+    private static void ReportNonUiThread(string property, [CallerMemberName] string methodName = "")
+    {
+        uiThread ??= Application.Current.Dispatcher.Thread;
+        var currentThread = Thread.CurrentThread;
+
+        if (currentThread == uiThread)
+        {
+            return;
+        }
+
+        var exception = new CustomStackTraceException($"Do not call {methodName}() from a thread other than the UI thread. Calling for property '{property ?? "{ property name null }"}'. (Calling from '{currentThread.Name ?? "{ name null }"}' @{currentThread.ManagedThreadId})");
+        
+#if DEBUG
+        throw exception;
+#else
+        exception.GenerateStackTrace();
```

Comment:
Won't this cause any problems in runtime? Not crashing due to invalid thread seems like a dangerous thing

Reply:
Well, we shouldn't be able to implement wrong threads anymore as it will throw in debug

And sometimes another thread can update it without it crashing as far as I understand it so maybe let's not crash for something that doesn't need to be crashed